server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -904,11 +928,22 @@ private void doExport() {
             final Callable<T> capturedExport;
             synchronized (this) {
                 capturedExport = exportMain;
-                if (state != ExportNotification.State.QUEUED || session.isExpired() || capturedExport == null) {
-                    return; // had a cancel race with client
+                // check for some sort of cancel race with client
+                if (state != ExportNotification.State.QUEUED
+                        || session.isExpired()
+                        || capturedExport == null
+                        || !tryRetainReference()) {
+                    if (errorHandler != null) {
+                        // fulfill promise to client that we will notify them whether we succeed or fail
+                        assignErrorId();
+                        errorHandler.onError(ExportNotification.State.CANCELLED, errorId, null, null);
```

Comment:
1. Should `assignErrorId()` be idempotent? If we've already assigned an ID, should we still be calling `onError`?
2. `null, null`?

Reply:
Should `dependentHandle` be called something more like `failedDependencyHandle`?